### PR TITLE
Dom mutations to support inline edit

### PR DESCRIPTION
**Pull request by gazzdingo**
Still need to tidy up the code

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 export const validAttributeName = /^[a-zA-Z:_][a-zA-Z0-9:_.-]*$/;
 const nullController: MutationController = {
   revert: () => {},
+  stop: () => {},
```
Comment (gazzdingo) [inline src/index.ts]: do not need anymore

File: src/index.ts [hunk 2]
```diff
@@ -354,43 +355,68 @@ function refreshAllElementSets() {
...
   mutations.add(m);
-  // run refresh on init to establish list of elements associated w/ mutation
   refreshElementsSet(m);
+  // run refresh on init to establish list of elements associated w/ mutation
   return {
     revert: () => {
       revertMutation(m);
     },
+    stop: () => {
+       disconnectGlobalObserver()
```
Comment (gazzdingo) [inline src/index.ts]: do not need anymore

File: src/index.ts [hunk 3]
```diff
@@ -510,6 +536,8 @@ function declarative({
 
 export type MutationController = {
   revert: () => void;
+  stop: () => void;
```
Comment (gazzdingo) [inline src/index.ts]: do not need anymore

File: src/index.ts [hunk 4]
```diff
@@ -376,17 +387,26 @@ export function connectGlobalObserver() {
...
+};
+
 function newMutation(m: Mutation): MutationController {
   // Not in a browser
   if (typeof document === 'undefined') return nullController;
   // add to global index of mutations
   mutations.add(m);
-  // run refresh on init to establish list of elements associated w/ mutation
   refreshElementsSet(m);
+  // run refresh on init to establish list of elements associated w/ mutation
```
Comment (bryce-fitzsimons) [inline src/index.ts]:
```suggestion
  // run refresh on init to establish list of elements associated w/ mutation
  refreshElementsSet(m);
```

**Pull request merged**